### DISCUSSION TITLE
Changelog in root directory

### DISCUSSION THREAD

**Discussion by thejasonxie** [ghiscoding/lerna-lite#661] [Q&A] [answered]
Hi, I have an issue generating the CHANGELOG.md in the root directory when `pnpm exec lerna version --conventional-commits` is used. Only changelog within the packages directory are generated. I am using pnpm workspaces

lerna.json:
```json
{
  "$schema": "node_modules/@lerna-lite/cli/schemas/lerna-schema.json",
  "version": "independent",
  "npmClient": "pnpm",
  "command": {
    "publish": {
      "conventionalCommits": true,
      "message": "chore(release): publish new version %v",
      "registry": "https://npm.pkg.github.com"
    }
  },
  "packages": ["packages/*"]
}
```
pnpm-workspace.yaml
```yaml
packages:
  # all packages in direct subdirs of packages/
  - "packages/*"
  # exclude packages that are inside test directories
  - "!**/test/**"

```

I'm fairly new to lerna in general so I'm not sure if its a bug so I opened a discussion instead of an issue. Thank you for your work with lerna-lite!

**Comment by ghiscoding** [comment]
I'm not sure why you're using `pnpm exec`, is there a reason why you didn't simply use a regular npm script? I always use npm scripts and it works fine, I would suggest to watch the youtube video that I've referenced on the main readme page
https://github.com/lerna-lite/lerna-lite#good-lerna-tutorials--references

Lerna-Lite also uses pnpm, so you could also take a look at the config that I use in this project. You should probably use npm scripts like I mentioned.

**Reply by thejasonxie** [reply]
I used `pnpm exec` because I didn't want to install it globaly and i also have pnpm usage enforced. I'm wondering if its because i set "version" to "independent" in the lerna.json file? 

This [repo](https://github.com/wireapp/wire-web-packages) that uses lerna-lite uses independent version and don't have a changelog in the root.

**Reply by thejasonxie** [reply, marked as answer]
That was the issue. I set the version to 0.0.0 in lerna.json and that generate a changelog in the root directory.

On the side note, i noticed that the %v in the command.publish.message doesn't work properly when "version" is set to "independent"
![Screenshot 2023-07-02 at 9 39 06 PM](https://github.com/lerna-lite/lerna-lite/assets/34463193/f6d82b30-5cf1-4a5b-931d-a186b6c1f849)

when I used a version number instead, it resolved it.

**Reply by ghiscoding** [reply]
I don't think %v can be used with independent, I'm not sure since I didn't create Lerna, I only forked it to create Lerna-Lite and I never used the independent mode myself... and I just checked the documentation and it does say that it only works with fixed version mode (which mean that it doesn't work with `independent` mode), I highlighted in bold the section that concerns your use case

https://github.com/lerna-lite/lerna-lite/tree/main/packages/version#--message-msg

> If the message contains `%s`, it will be replaced with the new global version version number prefixed with a "v". If the message contains `%v`, it will be replaced with the new global version version number without the leading "v". **Note that this placeholder interpolation only applies when using the default "fixed" versioning mode**, as there is no "global" version to interpolate when versioning independently.
